src/remediations/fifi.js

```diff
@@ -46,7 +46,28 @@ async function fetchRHCClientId (systems) {
     const systemIds = _.map(systems, system => { return system.id; });
     const systemProfileDetails = await inventory.getSystemProfileBatch(systemIds);
 
-    _.forEach(systems, system => system.rhc_client = systemProfileDetails[system.id].system_profile.rhc_client_id);
+    _.forEach(systems, system => {
+        system.rhc_client = systemProfileDetails[system.id].system_profile.rhc_client_id;
+        system.marketplace = systemProfileDetails[system.id].system_profile.is_marketplace;
+    });
+}
+
+function rhcAdditionalSorting (partitionedSystems) {
```

Comment:
This is pretty cumbersome. Instead of having a function that does multiple things and then returns an array of arrays it would be better do the filtering directly where needed.